src/main/java/de/bonndan/nivio/assessment/StatusValue.java

```diff
@@ -54,23 +63,26 @@
     /**
      * Creates a summary status value.
      *
-     * @param field composed field name
-     * @param max   max/highest status value
+     * @param fieldName composed field name
+     * @param maxValue   max/highest status value
      * @return summary
      */
-    public static StatusValue summary(@NonNull String field, @NonNull StatusValue max) {
-
-        if (StringUtils.isEmpty(field)) {
-            throw new IllegalArgumentException("Status value has no field");
+    @NonNull
+    public static StatusValue summary(@NonNull String fieldName, @NonNull StatusValue maxValue) {
+        if (StringUtils.isEmpty(fieldName)) {
+            throw new IllegalArgumentException("Status value has no field name.");
         }
-        return new StatusValue(field, max.getStatus(), max.getMessage(), max.getField());
+        if (StringUtils.isEmpty(maxValue)) {
+            throw new IllegalArgumentException("Status value has no max status value.");
+        }
+        return new StatusValue(fieldName, maxValue.getStatus(), maxValue.getMessage(), maxValue.getField());
     }
 
     public StatusValue(@NonNull String field, @Nullable Status status, @Nullable String message) {
         if (StringUtils.isEmpty(field)) {
             throw new IllegalArgumentException("Status value has no field");
         }
-        this.field = field;
+        this.field = Objects.requireNonNull(field);
```

Comment:
StringUtils.isEmpty also handles null. This is why I like it.

Reply:
Yes, I know that and use it too for null and empty checks. I don't know why I did that 😄